### PR TITLE
Use Docker bridged networking on platforms other than Linux.

### PR DESCRIPTION
**Pull request by zvin**
Connects-To: #110

Change-Type: patch
---- Autogenerated Waffleboard Connection: Connects to #110

### CODE REVIEW

File: src/preload.py [hunk 1]
```diff
@@ -189,12 +188,30 @@ def start_docker_daemon(storage_driver, mountpoint):
...
 @contextmanager
 def docker_context_manager(storage_driver, mountpoint):
-    running_dockerd = start_docker_daemon(storage_driver, mountpoint)
+    docker_dir = fix_rce_docker(mountpoint)
+    # If we don't remove <part6>/<docker|rce>/network/files/local-kv.db and the
+    # preload container was started with bridged networking, the following
+    # dockerd is not reachable from the host.
+    local_kv_db_path = "{}/network/files/local-kv.db".format(docker_dir)
+    local_kv_db_content = read_file(local_kv_db_path)
+    os.remove(local_kv_db_path)
```
Comment (zvin) [inline src/preload.py]: @petrosagg do you know why this file prevents in inner dockerd from being reachable from the host if the container is started with bridged networking?

File: lib/preload.js [hunk 2]
```diff
@@ -88,8 +87,26 @@ const createContainer = (docker, options) => {
...
+      containerOptions.ExposedPorts = {}
+      containerOptions.ExposedPorts[`${options.dockerPort}/tcp`] = {}
+      containerOptions.HostConfig.PortBindings = {}
+      containerOptions.HostConfig.PortBindings[`${options.dockerPort}/tcp`] = [{
+        HostPort: `${options.dockerPort}`,
+        HostIp: ''
+      }]
+    }
+    containerOptions.HostConfig.NetworkMode = 'bridge'
+  }
```
Comment (CameronDiver) [inline lib/preload.js]: I'm assuming this is tested on both mac+linux ?

Comment (zvin) [reply lib/preload.js]: Actually I only tested on Linux.
This PR is supposed to make preload work on macs.
Could you please test it on a mac? I don't have one.

Comment (CameronDiver) [reply lib/preload.js]: @shaunmulligan are you able to help us here please?

**Comment by shaunmulligan** [comment]
I ran the `resin-preload` binary directly and it looks to have worked correctly. I am running MacOS Sierra 10.12.6
```
Building preloader image...

Building Docker preloader image. [========================] 100%

Preloading...

Fetching container size for moma/72f92753379b00abd20c479c1eb4ed37a3b0c6c0 [========================] 100%
Leaving splash image alone
Expanding image size by 593.71 MiB
Expanding extended partition 4 and logical partition 6 to the end of the disk image.
Expanding ext filesystem
Using /dev/loop0
Resizing filesystem
e2fsck: File system OK
Waiting for Docker to start...
Docker started

Pulling image [========================] 100%

[OK] Done.
```
and the device came up with the app. This was with `Resin OS 2.3.0+rev1 (dev)` . 

So I think for MacOS this looks good 👍

**Comment by zvin** [comment]
thanks @shaunmulligan

**Pull request merged**